src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -147,17 +153,29 @@ public void registerIdentityChangeListener(Consumer<Principal> listener) {
         DirContext dirContext = null;
         try {
             dirContext = obtainContext();
-            EventContext eventContext = (EventContext) dirContext.lookup("");
-            eventContext.addNamingListener("", EventContext.SUBTREE_SCOPE, new ServerNotificationListener(listener));
+            innerRegisterIdentityChangeListener(dirContext,listener);
         } catch (Exception cause) {
-            throw log.ldapRealmFailedRegisterListener(cause);
+            //either connection died or realm not available during boot
+            synchronized (this.listenersPendingRegistration) {
+                //we need to wait, lets store
+                listenersPendingRegistration.add(listener);
+            }
+            log.ldapRealmDeferRegistration(this.name);
+            if(log.isDebugEnabled()){
+                log.debug("Listener registration failure: ", cause);
+            }
         } finally {
             if (dirContext != null) {
                 closeContext(dirContext);
             }
         }
     }
 
+    private void innerRegisterIdentityChangeListener(final DirContext dirContext, final Consumer<Principal> listener) throws NamingException {
```

Comment:
The problem is the prefixing itself - it is bad practise called Hungarien notation. You should not put such information into name when you are able to do it on type system level using `private` keyword and similar.
http://stackoverflow.com/questions/111933/why-shouldnt-i-use-hungarian-notation